.github/workflows/pull-request.yml

```
@@ -0,0 +1,1162 @@
+        if: failure()
+        run: pnpm format:fix
+
+      - name: Check for type errors
+        if: steps.changed-files.outputs.only_changed != 'true'
+        run: pnpm typecheck
+
+      - name: Check for linting errors in modified files
+        if: steps.changed-files.outputs.only_changed != 'true'
+        env:
+          CHANGED_FILES: ${{ steps.changed-files.outputs.all_changed_files }}
+        run: pnpm exec eslint ${CHANGED_FILES}
```

actionlint (1.7.9)

shellcheck reported issue in this script: SC2086:info:1:18: Double quote to prevent globbing and word splitting

(shellcheck)